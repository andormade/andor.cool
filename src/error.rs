use std::fmt;
use std::io;

#[derive(Debug)]
pub enum Error {
    /// IO operation error
    Io(io::Error),
    /// Error that occurs during handlebars processing
    Handlebars(String),
}

impl fmt::Display for Error {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match self {
            Error::Io(err) => write!(f, "IO error: {}", err),
            Error::Handlebars(msg) => write!(f, "Handlebars error: {}", msg),
        }
    }
}

impl std::error::Error for Error {
    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
        match self {
            Error::Io(err) => Some(err),
            _ => None,
        }
    }
}

impl From<io::Error> for Error {
    fn from(err: io::Error) -> Self {
        Error::Io(err)
    }
}

impl From<String> for Error {
    fn from(err: String) -> Self {
        Error::Handlebars(err)
    }
}

pub type Result<T> = std::result::Result<T, Error>; 