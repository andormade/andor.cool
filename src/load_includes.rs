use std::collections::HashMap;
use std::fs;
use std::path::Path;
use crate::error::MyError;

pub fn load_liquid_includes(dir_path: &str) -> Result<HashMap<String, String>, MyError> {
    let path = Path::new(dir_path);
    let mut templates = HashMap::new();

    for entry in fs::read_dir(path)? {
        let entry = entry?;
        let path = entry.path();
        if path.is_file() && path.extension().and_then(|ext| ext.to_str()) == Some("liquid") {
            if let Some(filename) = path.file_name().and_then(|name| name.to_str()) {
                let contents = fs::read_to_string(&path)?;
                templates.insert(filename.to_string(), contents);
            }
        }
    }

    Ok(templates)
}

#[cfg(test)]
mod tests {
    use super::*;
    use insta::assert_snapshot;
    use serde_json;
    use std::collections::BTreeMap;

    #[test]
    fn test_load_liquid_includes() {
        // Update test to handle Result
        let templates = load_liquid_includes("./_includes").unwrap_or_default();
        let sorted_templates: BTreeMap<_, _> = templates.into_iter().collect();
        let templates_json = serde_json::to_string_pretty(&sorted_templates).unwrap();
        // Use assert_snapshot! to compare the output against the stored snapshot
        assert_snapshot!(templates_json);
    }
}
