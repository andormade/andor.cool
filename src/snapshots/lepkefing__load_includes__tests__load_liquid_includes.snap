---
source: src/load_includes.rs
expression: templates_json
---
{
  "candystore.liquid": "<a href=\"https://static.llllllllllll.com/andor/images/{{ file }}_original.jpg\"><img src=\"https://static.llllllllllll.com/andor/images/{{ file }}.jpg\" alt=\"{{ title }}\" title=\"{{ title }}\" loading=\"lazy\" /></a>\n",
  "list_item.liquid": "<li><a href=\"/posts/{{slug}}\">{{title}}</a></li>",
  "video.liquid": "<div class=\"video-container-container\">\n    <video width=\"100%\" controls preload=\"metadata\">\n        <source src=\"https://static.llllllllllll.com/andor/videos/{{id}}\" type=\"video/mp4\">\n    </video>\n</div>",
  "vimeo.liquid": "<div class=\"video-container-container\">\n\t<div class=\"video-container video-container-16-9\">\n\t\t<iframe src=\"https://player.vimeo.com/video/{{id}}\" frameborder=\"0\" allowfullscreen></iframe>\n\t</div>\n</div>",
  "youtube.liquid": "<div class=\"video-container video-container-16-9\">\n\t<iframe src=\"https://www.youtube-nocookie.com/embed/{{id}}\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen></iframe>\n</div>"
}
